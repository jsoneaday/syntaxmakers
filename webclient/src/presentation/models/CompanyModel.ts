import { Key } from "react";
import { Company } from "../../domain/repository/CompanyRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type CompanyModel = {
  key: Key;
  id: string;
  updatedAt: string;
  name: string;
};
export default CompanyModel;

export function convert(company: Company): CompanyModel {
  return {
    key: uuidv4(),
    id: company.id,
    updatedAt: company.updatedAt,
    name: company.name,
  };
}
