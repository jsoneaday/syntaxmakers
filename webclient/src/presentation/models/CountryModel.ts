import { Key } from "react";
import { Country } from "../../domain/repository/CountryRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type CountryModel = {
  key: Key;
  id: string;
  updatedAt: string;
  name: string;
};
export default CountryModel;

export function convert(country: Country): CountryModel {
  return {
    key: uuidv4(),
    id: country.id,
    updatedAt: country.updatedAt,
    name: country.name,
  };
}
