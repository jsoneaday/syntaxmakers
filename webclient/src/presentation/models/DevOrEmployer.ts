export enum DevOrEmployer {
  Developer = "Developer",
  Employer = "Employer",
}
