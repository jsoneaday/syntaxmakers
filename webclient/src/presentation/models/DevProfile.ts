import { Key } from "react";
import { Developer } from "../../domain/repository/DeveloperRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type DevProfile = {
  key: Key;
  id: string; // bigint is not serializable by Redux
  updatedAt: string;
  userName: string;
  fullName: string;
  email: string;
  primaryLangId: number;
  secondaryLangId?: number | null;
  accessToken?: string | null;
};
export default DevProfile;

export function convert(dev: Developer, accessToken?: string): DevProfile {
  return {
    key: uuidv4(),
    id: dev.id, // bigint is not serializable by Redux
    updatedAt: dev.updatedAt,
    userName: dev.userName,
    fullName: dev.fullName,
    email: dev.email,
    primaryLangId: dev.primaryLangId,
    secondaryLangId: dev.secondaryLangId,
    accessToken,
  };
}
