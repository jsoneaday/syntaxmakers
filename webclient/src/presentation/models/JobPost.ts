import { Key } from "react";
import { KeyItem } from "../common/utils";

export default class JobPost implements KeyItem {
  constructor(
    public key: Key,
    public id: bigint,
    public updatedAt: string,
    public title: string,
    public description: string,
    public companyName: string,
    public isRemote: boolean,
    public primaryLangName: string,
    public secondaryLangName: string,
    public industryName: String,
    public salary: string,
    public logo?: Blob,
    public countryName?: string
  ) {}
}
