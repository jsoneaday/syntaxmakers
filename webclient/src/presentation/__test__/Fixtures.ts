import DevProfile from "../models/DevProfile";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import { faker } from "@faker-js/faker";
import { getFakeFullName } from "../../domain/TestUtils";

export const defaultDevProfile: DevProfile = {
  key: uuidv4(),
  id: "1",
  updatedAt: new Date().toISOString(),
  userName: "ronaldo", //faker.internet.userName(),
  fullName: getFakeFullName(),
  email: faker.internet.email(),
  primaryLangId: "1",
};
