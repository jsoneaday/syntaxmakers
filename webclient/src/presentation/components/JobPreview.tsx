import JobPost from "../models/JobPost";
import "../theme//post_preview.css";

interface JobPreviewProps {
  jobPost: JobPost;
  isSmall: boolean;
}

export default function JobPreview({ jobPost, isSmall }: JobPreviewProps) {
  const smallTitleFont = isSmall
    ? "title-font preview-small-title-font"
    : "title-font";

  const smallSubTitleFont = isSmall
    ? "sub-title-font preview-small-sub-title-font"
    : "sub-title-font";

  const smallNormalFont = isSmall
    ? "normal-font preview-small-normal-font"
    : "normal-font";

  const icon = isSmall ? null : (
    <img
      className="preview-icon"
      src={jobPost.companyLogo ? URL.createObjectURL(jobPost.companyLogo) : ""}
    />
  );

  const timestamp = isSmall ? null : (
    <div className="small-font preview-timestamp">{jobPost.updatedAt}</div>
  );

  return (
    <div className="post-preview-container">
      {icon}
      <div className="preview-content">
        <div className={smallTitleFont}>{jobPost.title}</div>
        <div className={smallSubTitleFont}>{jobPost.companyName}</div>
        <div className={smallNormalFont}>
          {jobPost.isRemote ? "Remote" : jobPost.countryName}
        </div>
        <div className={"normal-font preview-salary"}>
          Base Salary: <i>{`$${jobPost.salary}`}</i>
        </div>
        {timestamp}
      </div>
    </div>
  );
}
