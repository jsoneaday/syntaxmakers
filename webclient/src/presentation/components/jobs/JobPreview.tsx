import JobPost from "../../models/JobPost";
import "../../theme/post_preview.css";

interface JobPreviewProps {
  jobPost: JobPost;
  isSmall: boolean;
}

export default function JobPreview({ jobPost, isSmall }: JobPreviewProps) {
  const smallTitleFont = isSmall
    ? "title-font preview-small-title-font"
    : "title-font";

  const smallSubTitleFont = isSmall
    ? "sub-title-font preview-small-sub-title-font"
    : "sub-title-font";

  const smallNormalFont = isSmall
    ? "normal-font preview-small-normal-font"
    : "normal-font";

  const timestamp = isSmall ? null : (
    <div className="small-font preview-timestamp">{jobPost.updatedAt}</div>
  );

  return (
    <div className="post-preview-container">
      <div className="preview-content">
        <div className={smallTitleFont}>{jobPost.title}</div>

        <div style={{ marginBottom: ".25em" }} className={smallSubTitleFont}>
          {jobPost.companyName}
        </div>
        <div style={{ padding: ".25em" }} className={smallNormalFont}>
          {jobPost.isRemote ? "Remote" : jobPost.countryName}
        </div>

        <div style={{ padding: ".25em" }}>
          <div className={"normal-font preview-item"}>
            Primary Language: <i>{`${jobPost.primaryLangName}`}</i>
          </div>
          <div className={"normal-font preview-item"}>
            Secondary Language: <i>{`${jobPost.secondaryLangName}`}</i>
          </div>
          <div className={"normal-font preview-item"}>
            Base Salary: <i>{`${jobPost.salary}`}</i>
          </div>
        </div>
        <div style={{ paddingLeft: ".25em" }}>{timestamp}</div>
      </div>
    </div>
  );
}
