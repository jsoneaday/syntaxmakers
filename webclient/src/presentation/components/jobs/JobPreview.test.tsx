import configureStore from "redux-mock-store";
import { defaultEmpProfile } from "../../__test__/Fixtures";
import { DevOrEmployer } from "../../models/DevOrEmployer";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { RouterProvider, createBrowserRouter } from "react-router-dom";
import JobPreview from "./JobPreview";
import JobPost from "../../models/JobPost";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import { faker } from "@faker-js/faker";

const title = faker.string.sample(10);
const employerName = faker.internet.displayName();
const isRemote = true;
const companyName = faker.company.name();
const primaryLangName = "JavaScript";
const secondaryLangName = "C#";
const industryName = "Finance";
const salary = "200000";
const countryName = undefined;

const jobPost: JobPost = {
  key: uuidv4(),
  id: 1,
  updatedAt: "2022/01/03T22:21:02",
  title,
  description: '[{"type":"paragraph","children":[{"text":"Testing Text"}]}]',
  employerId: 1,
  employerName,
  companyId: 1,
  companyName,
  isRemote,
  primaryLangId: 1,
  primaryLangName,
  secondaryLangId: 1,
  secondaryLangName,
  industryId: 1,
  industryName,
  salaryId: 1,
  salary,
  countryName,
};

const router = createBrowserRouter([
  {
    path: "/",
    element: <JobPreview jobPost={jobPost} isSmall={false} />,
  },
]);

describe("Test JobPreview", () => {
  it("matches snapshot", async () => {
    const mockStore = configureStore();
    const store = mockStore({
      profile: defaultEmpProfile,
      devOrEmployer: DevOrEmployer.Employer,
    });

    const result = render(
      <Provider store={store}>
        <RouterProvider router={router} />
      </Provider>
    );

    screen.getByText(title);
    screen.getByText(companyName);
    screen.getByText("Remote");
    screen.getByText(salary);
    expect(result).toMatchSnapshot();
  });
});
