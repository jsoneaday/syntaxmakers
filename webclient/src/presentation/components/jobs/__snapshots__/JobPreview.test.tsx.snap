// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test JobPreview matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="post-preview-container"
      >
        <img
          class="preview-icon"
          src=""
        />
        <div
          class="preview-content"
        >
          <div
            class="title-font"
          >
            {j[&lt;_\`{C:.
          </div>
          <div
            class="sub-title-font"
          >
            Watsica, Rippin and Fisher
          </div>
          <div
            class="normal-font"
          >
            Remote
          </div>
          <div
            class="normal-font preview-salary"
          >
            Base Salary: 
            <i>
              200000
            </i>
          </div>
          <div
            class="small-font preview-timestamp"
          >
            2022/01/03T22:21:02
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="post-preview-container"
    >
      <img
        class="preview-icon"
        src=""
      />
      <div
        class="preview-content"
      >
        <div
          class="title-font"
        >
          {j[&lt;_\`{C:.
        </div>
        <div
          class="sub-title-font"
        >
          Watsica, Rippin and Fisher
        </div>
        <div
          class="normal-font"
        >
          Remote
        </div>
        <div
          class="normal-font preview-salary"
        >
          Base Salary: 
          <i>
            200000
          </i>
        </div>
        <div
          class="small-font preview-timestamp"
        >
          2022/01/03T22:21:02
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
